v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/model/helpers/MappingMapper.java

```diff
@@ -15,294 +15,493 @@
-import com.google.cloud.teleport.v2.neo4j.model.enums.FragmentType;
-import com.google.cloud.teleport.v2.neo4j.model.enums.PropertyType;
-import com.google.cloud.teleport.v2.neo4j.model.enums.RoleType;
-import com.google.cloud.teleport.v2.neo4j.model.enums.TargetType;
-import com.google.cloud.teleport.v2.neo4j.model.job.FieldNameTuple;
-import com.google.cloud.teleport.v2.neo4j.model.job.Mapping;
-import com.google.cloud.teleport.v2.neo4j.model.job.Target;
-import com.google.cloud.teleport.v2.neo4j.utils.ModelUtils;
+import static com.google.cloud.teleport.v2.neo4j.model.helpers.JsonObjects.getBooleanOrDefault;
+import static com.google.cloud.teleport.v2.neo4j.model.helpers.JsonObjects.getStringOrDefault;
+import static com.google.cloud.teleport.v2.neo4j.model.helpers.JsonObjects.getStringOrNull;
+import static com.google.cloud.teleport.v2.neo4j.model.helpers.MappingVisitor.visit;
+import static com.google.cloud.teleport.v2.neo4j.model.helpers.SourceMapper.DEFAULT_SOURCE_NAME;
+import static org.neo4j.importer.v1.targets.PropertyType.BOOLEAN;
+import static org.neo4j.importer.v1.targets.PropertyType.BYTE_ARRAY;
+import static org.neo4j.importer.v1.targets.PropertyType.FLOAT;
+import static org.neo4j.importer.v1.targets.PropertyType.INTEGER;
+import static org.neo4j.importer.v1.targets.PropertyType.POINT;
+import static org.neo4j.importer.v1.targets.PropertyType.STRING;
+import static org.neo4j.importer.v1.targets.PropertyType.ZONED_DATETIME;
+
 import java.util.ArrayList;
-import java.util.Iterator;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
 import java.util.List;
-import org.apache.commons.lang3.StringUtils;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
 import org.json.JSONArray;
 import org.json.JSONObject;
+import org.neo4j.importer.v1.targets.NodeExistenceConstraint;
+import org.neo4j.importer.v1.targets.NodeKeyConstraint;
+import org.neo4j.importer.v1.targets.NodeRangeIndex;
+import org.neo4j.importer.v1.targets.NodeSchema;
+import org.neo4j.importer.v1.targets.NodeTarget;
+import org.neo4j.importer.v1.targets.NodeUniqueConstraint;
+import org.neo4j.importer.v1.targets.PropertyMapping;
+import org.neo4j.importer.v1.targets.RelationshipExistenceConstraint;
+import org.neo4j.importer.v1.targets.RelationshipKeyConstraint;
+import org.neo4j.importer.v1.targets.RelationshipRangeIndex;
+import org.neo4j.importer.v1.targets.RelationshipSchema;
+import org.neo4j.importer.v1.targets.RelationshipUniqueConstraint;
+import org.neo4j.importer.v1.targets.Target;
+import org.neo4j.importer.v1.targets.WriteMode;
 
-/** Helper object for parsing field mappings (ie. strings, indexed, longs, etc.). */
-public class MappingMapper {
+/**
+ * Helper object for parsing legacy json for property mappings, schema, labels and types.
+ *
+ * @deprecated use the current JSON format instead
+ */
+@Deprecated
+class MappingMapper {
```

Comment:
would `LegacyToNewSpecMapper` be a better name, wdyt?

Reply:
I did not rename the existing classes as I thought it would maybe help navigating from the old codebase to the new one... not sure that's a good enough reason to keep the old names around though.